src/Microsoft.AspNetCore.Server.Kestrel/Infrastructure/Constants.cs

```diff
@@ -28,32 +28,36 @@ internal class Constants
 
         private static int? GetECONNRESET()
         {
-            switch (PlatformServices.Default.Runtime.OperatingSystemPlatform)
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
-                case Platform.Windows:
-                    return -4077;
-                case Platform.Linux:
-                    return -104;
-                case Platform.Darwin:
-                    return -54;
-                default:
-                    return null;
+               return -4077;
             }
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
```

Comment:
In this case I don't have any preference just wanted to know what your thoughts are. In more complicated cases I often times find `else if`s tricky - especially when they introduce new variables and suddenly you need to work on multiple set intersections many of which are not covered by any of the clauses/conditions. In such cases I prefer (if possible and not too nested) an `if` nested in `else` than `else if`. .